### ISSUE TITLE
[BUG] Compile error MSVC when Unicode is enabled

### ISSUE THREAD

**Issue by JessyDL** [google/benchmark#1515]
**Describe the bug**
On 1.7.1 and HEAD with MSVC and `UNICODE` enabled a compile error is triggered on line: `src/sysinfo.cc:428:15` (error C2664)

I noticed this code was recently changed in PR #1474. My conversion knowledge is lacking here, but I've never seen a `std::string` initialized using `wchar_t*`, so my gut feeling is there is something off about this code. Aside from that `MultiByteToWideChar` does not have a version that accepts `wchar` from inspecting the windows headers (and docs), while `TCHAR` is set to `wchar_t` when `UNICODE` is enabled.

Undoing the PR fixes the issue locally for me.

edit: It seems the code that PR replaced was used to transform a wide string to a multibyte one (`std::wstring_convert::to_bytes()`). The code that was written in that PR however did the opposite (tried transforming a wide string into a wide string), and ended up trying to initialize a string with that result (which shouldn't be possible as `std::string` lacks a `wchar_t*` constructor). I'm fairly confused at the intent here. I've prepared a PR that implements the expected behaviour using the win32 API (see my comment in this issue).

**System**
Which OS, compiler, and compiler version are you using:
  - OS: Windows 10
  - Compiler and version: VC++ 17.4 (_MSC_VER 1934)

**To reproduce**
1. set up the project following the step guides of the readme up to the building step
2. run `cmake --build "build" --config Release -- /p:CharacterSet=Unicode`

The second step enables UNICODE for msbuild.

**Expected behavior**
No compile errors

**Additional context**

Error from the compile log (stripped identifying information)
```
8>sysinfo.cc
8>src\sysinfo.cc(428,15): error C2664: 'int MultiByteToWideChar(UINT,DWORD,LPCCH,int,LPWSTR,int)': cannot convert argument 3 from 'TCHAR [16]' to 'LPCCH'
8>src\sysinfo.cc(428,66): message : Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or parenthesized function-style cast
8>Windows Kits\10\Include\10.0.19041.0\um\stringapiset.h(126,1): message : see declaration of 'MultiByteToWideChar'
8>src\sysinfo.cc(428,15): message : while trying to match the argument list '(int, int, TCHAR [16], DWORD, std::_Vector_iterator<std::_Vector_val<std::_Simple_types<_Elem>>>, int)'
8>        with
8>        [
8>            _Elem=wchar_t
8>        ]
8>src\sysinfo.cc(434,11): error C2664: 'int MultiByteToWideChar(UINT,DWORD,LPCCH,int,LPWSTR,int)': cannot convert argument 3 from 'TCHAR [16]' to 'LPCCH'
8>src\sysinfo.cc(434,62): message : Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or parenthesized function-style cast
8>Windows Kits\10\Include\10.0.19041.0\um\stringapiset.h(126,1): message : see declaration of 'MultiByteToWideChar'
8>src\sysinfo.cc(434,11): message : while trying to match the argument list '(int, int, TCHAR [16], DWORD, std::_Vector_iterator<std::_Vector_val<std::_Simple_types<_Elem>>>, unsigned __int64)'
8>        with
8>        [
8>            _Elem=wchar_t
8>        ]
8>src\sysinfo.cc(438,20): error C2440: '<function-style-cast>': cannot convert from '_Ty *' to 'std::string'
8>        with
8>        [
8>            _Ty=wchar_t
8>        ]
8>sysinfo.cc(438,20): message : No constructor could take the source type, or constructor overload resolution was ambiguous
```

**Comment by JessyDL** [comment]
I've made a PR that resolves this issue on my end. See here #1516

**Comment by JessyDL** [comment]
Issue was resolved in PR #1516, and is now merged
